drive-contents/code.py

```diff
@@ -1,136 +1,30 @@
-
-    def _wearable_(self):
-        view = self.system_view((left, top), self.dims)
-        self.swipeup((0, 240), (self.width, SWIPE_HEIGHT))
-        self.swipedown((0, 0), (self.width, SWIPE_HEIGHT))
-        self.swipeleft((240, 0), (SWIPE_WIDTH, self.height))
-        self.swiperight((0, 0), (SWIPE_WIDTH, self.height))
-
-
-swipeup = WatchRoot.swipeup
-swipedown = WatchRoot.swipedown
-swipeleft = WatchRoot.swipeleft
-swiperight = WatchRoot.swiperight
-# print(f"swipeup={swipeup}, swipedown={swipedown}")
-
-swipeup._start_coord_ = lambda coord: None  # print('swipeup._start_coord_', coord)
-swipeup._update_coord_ = lambda _: None
-swipeup._last_coord_ = lambda coord: (
-    WatchRoot.system_view.swipe_up() if coord[1] <= 0 else None
-)
-
-swipedown._start_coord_ = lambda _: None
-swipedown._update_coord_ = lambda _: None
-swipedown._last_coord_ = lambda coord: (
-    WatchRoot.system_view.swipe_down() if coord[1] >= SWIPE_HEIGHT else None
-)
-print(gc.mem_free())
-
-gc.collect()
-print(WatchRoot)
-WatchRoot._superior_._std_startup_()
-
-g = WatchRoot._superior_._group
-print(g, g[0])
-gc.collect()
-
-# a = Rect(fill=color.red)
-# b = Rect(fill=color.blue)
-# a._id_ = "'red'"
-# b._id_ = "'blue'"
+from hardware import drivers
+
+# # try to import the hardware, this check may not be needed anymore
+# try:
+#     from hardware import drivers
+# except Exception as err:
+#     if "SCK in use" in err.args or "lock timed out" in err.args[0]:
+#         import oh_shit
 #
-# WatchRoot.system_view.appview.app = a
-# WatchRoot.system_view.appview.app = b
-# print(WatchRoot.system_view.appview._nested_)
-if __name__ == "__main__":
+#         oh_shit.reset_countdown(30, err)
+#     raise err
 
-    try:
-        gc.collect()
-        print(gc.mem_free())
-        while True:
-            gc.collect()
-            # print("loop:", gc.mem_free())
-            # _ in range(30):
-            system._refresh()
-            event_loop.loop()
-            display.refresh()
 
-    except Exception as err:
-        gc.enable()
-        print("exiting")
-        del WatchRoot, screen
-        gc.collect()
+if drivers._read_bat_percent() < 4:
+    from boot_options import low_battery
+else:
+    # show splash while loading watch_gui
+    from boot_options import splash_screen
 
-        sys.print_exception(err)
-        try:
-            with open("error.text", "w") as file:
-                sys.print_exception(err, file=file)
-        except:
-            pass
-        display.show(None)
-        display.refresh()
-        # hardware.deinit()
-        if isinstance(err, MemoryError):
-            oh_shit.reset_countdown(10, err)
-        else:
-            oh_shit.reset_countdown(30, err)
+    # init the gui system
+    from boot_options import watch_gui
 
-else:
-    display.auto_refresh = True
+    # clean up splash
+    sys.modules.pop(splash_screen.__name__)
+    del splash_screen
+
+    if __name__ == "__main__":
+        watch_gui.run()
```

Comment:
I'm still a little confused on how this worked but it might make more sense when I integrate the time setting boot option with this branch

Reply:
Which part do you find confusing? (like lines wise?)

Reply:
I guess I understand lines 26 and 27 but I don't get why 29 is a thing. I remember that for my special time setting boot to work I had to modify that line so that `"__main__"` was `"<nameOfFunction>"`

Reply:
Lines 26/27 are to free up any ram consumed by the splash screen. 
Though  `"__main__"` shouldn't need to be changed (it's a python idiom). If you want to know more [medium](https://medium.com/python-features/understanding-if-name-main-in-python-a37a3d4ab0c3)  does a decent explainer.